### ISSUE TITLE
README `--target` argument is outdated, and lacking `TokenAllowance` instructions for testing ERC20Wrapper

### ISSUE THREAD

**Issue by b-yap** [pendulum-chain/pendulum-ink-wrapper#37]
### `--target` command argument is outdated
solang version >= 0.3.1 does not accept _`substrate`_ argument anymore.

TODO: it should be **`--target polkadot`**
     
-----
     
### lacking `TokenAllowance` instructions for testing ERC20Wrapper
`ContractTrapped` error will occur when testing the functions using zombienet. It returns _6_(or `NoProvider`) after calling `chain_extension`.

TODO: Depending on which runtime, the chain_spec should add in its GenesisConfig:
```rust
token_allowance: amplitude_runtime::TokenAllowanceConfig {
   allowed_currencies: vec![CurrencyId::Native]
}
``` 
^ this means the `Native` currency is the only one supported. 
To be able to define`TokenAllowanceConfig` in the GenesisConfig, add **`Config<T>`** of `TokenAllowance` inside the `construct_runtime!` of the parachain. e.g.
```rust
construct_runtime! {
    ...
    ...
    TokenAllowance: orml_currencies_allowance_extension::{Pallet, Config<T>, Storage, Call, Event<T>} = 80,
    ...

}
```

**Comment by TorstenStueber** [comment]
@b-yap the first part is clear to me. For the second part (lacking `TokenAllowance`): isn't that a fix required in the runtime or client, i.e., in the pendulum repo instead of in the wrapper contract?

**Comment by prayagd** [comment]
@pendulum-chain/devs is this bug still relevant?

**Comment by ebma** [comment]
> For the second part (lacking TokenAllowance): isn't that a fix required in the runtime or client, i.e., in the pendulum repo instead of in the wrapper contract?

Yes kind of, but it wouldn't hurt to leave a note somewhere that having the currency in the list of `allowed_currencies` of the `TokenAllowance` pallet is a requirement and you will encounter errors if you don't take care of this before.
